modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryVersionLocalServiceImpl.java

```diff
@@ -159,6 +186,34 @@ public ObjectEntryVersion updateLatestObjectEntryVersion(
 			objectEntry.getVersion());
 	}
 
+	private boolean _checkMaximumObjectEntryVersions(long objectEntryId)
+		throws ConfigurationException {
+
+		boolean exceededVersionsNumber = false;
+
+		int versionsAmount = getObjectEntryVersionsCount(objectEntryId);
+
+		if (versionsAmount <= 0) {
+			return exceededVersionsNumber;
+		}
+
+		ObjectEntryVersionRetentionConfiguration
+			objectEntryVersionRetentionConfiguration =
+				_configurationProvider.getCompanyConfiguration(
```

Comment:
When we scope a configuration by Company, the user can configure it at the Instance and System configuration level, so I think you should implement code like [this](https://github.com/liferay/liferay-portal/blob/16bd880eae22d0c27f6babde69f5790294e5c78a/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryServiceImpl.java#L834-L847). Remember to test it.